keps/1833-metrics-for-local-queue/README.md

```diff
@@ -0,0 +1,167 @@
+management and to gain insights into their workloads within their limited scope of access.
+
+### Goals
+
+1. Introduce the API changes required to enable Local Queue metrics. 
+1. List the Prometheus metrics that would be exposed for Local Queues.  
+
+### Non-Goals
+
+1. Discuss the implementation details on where these metrics need to be collected in codebase.
+
+## Proposal
+
+The proposal extends to enable collection of metrics for local queues that would be useful
+for batch users and cluster administrators. 
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a batch user of Kueue, I want to access metrics for local queues running workloads restricted to my namespace so that 
+I can monitor and analyze the performance and trends of my workloads.
+
+#### Story 2
+
+As an administrator of Kueue, I want to enable batch users specific to a namespace to collect metrics for their workloads 
+within their namespace so that they can have visibility and insights into their own workload metrics.
+
+#### Story 3
+
+As an administrator of Kueue, I want to filter and gain insights on fine-grained metrics relevant to a local queue by 
+namespace for specific tenants so that I can effectively manage and optimize resource usage and performance for different tenants.
+
+## Design Details
+
+### API changes:
+
+The [Configuration API](https://github.com/kubernetes-sigs/kueue/blob/7ec127b05c8a0c8268e623de61914472dc5bff29/apis/config/v1beta1/configuration_types.go#L30)
+currently provides the ability to enable collection of metrics for cluster queues. This API will be extended to include options for enabling metrics 
+collection for local queues.
+
+The `ControllerMetrics` that contain the option to configure metrics, will be extended as follows:
+
+```go
+type ControllerManager struct {
+	...
+
+	// Metrics contains the controller metrics configuration
+	// +optional
+	Metrics ControllerMetrics `json:"metrics,omitempty"`
+	...
+}
+
+// ControllerMetrics defines the metrics configs.
+type ControllerMetrics struct {
+	...
+	
+	// EnableLocalQueueResources, if true the local queue resource usage and quotas
+	// metrics will be reported.
+	// +optional
+	EnableLocalQueueResources bool `json:"enableLocalQueueResources,omitempty"`
+	
+	// LocalQueueMetricOptions specifies the configuration options for local queue metrics.
+	LocalQueueMetricOptions *metricsOptions `json:"localQueueMetricOptions,omitempty"`
+}
+
+// metricsOptions defines the configuration options for local queue metrics.
+// If left empty, then metrics will expose for all local queues across namespaces.
+type metricsOptions struct {
+	// NamespaceSelector can be used to select namespaces in which the local queues should
+	// report metrics.
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	
+	// QueueSelector can be used to choose the local queues that need metrics to be collected. 
+	QueueSelector *metav1.LabelSelector `json:"queueSelector,omitempty"`
```

Comment:
What happens if the below situations?

1. only namespeceSelector is specified
2. only localQueueSelector is specified
3. both namespaceSelector and localQueueSelector are specified

Reply:
Had a predicate based filtering kind of approach in mind for this case. 

1. Only NS selector: All LocalQueues in the specific namespaces are selected. If it's nil, all the local queues across all namespaces are considered. 
2. LocalQueue selector: Local queues's matching the labels are selected across all namespaces. If not, all the local queues across namespaces are considered.
3. If both are specified: Both the selectors are applied to local queues (logical AND).

Had a Venn diagram in mind :)) Please let me know if I'm missing anything.

Reply:
Have added a configuration table for this knob in the KEP.